d3a_api_client/utils.py

```diff
@@ -210,4 +211,12 @@ def list_running_canary_networks_and_devices_with_live_data(domain_name):
         cn["uuid"]: cn["scenarioData"]["forecastStreamAreaMapping"]
         for cn in data["data"]["listCanaryNetworks"]["configurations"]
         if cn["resultsStatus"] == "running"
-    }
\ No newline at end of file
+    }
+
+
+def execute_function_util(function: callable, function_name):
```

Comment:
Is the `function_name` needed? What if we remove it and use something like `function.__name__`?

Reply:
Unfortunately I'm passing a closure lambda function, not the original one so this is not possible

Reply:
Edit: sorry, it's a lambda function. Ignore my previous comment :)